felix/dataplane/linux/int_dataplane.go

```diff
@@ -977,6 +979,30 @@ func ConfigureDefaultMTUs(hostMTU int, c *Config) {
 	}
 }
 
+func cleanUpIPIPAddrs() {
+	// If IPIP is not enabled, check to see if there is are addreses in the IPIP device and delete them if there are.
+	log.Debug("Checking if we need to clean up the IPIP device")
+	link, err := netlink.LinkByName("tunl0")
+	if err != nil {
+		if _, ok := err.(netlink.LinkNotFoundError); ok {
+			log.Debug("IPIP disabled and no IPIP device found")
+			return
+		}
+		log.WithError(err).Warn("IPIP disabled and failed to query IPIP device. Ignoring.")
```

Comment:
wonder if we need to retry in this case. It should be rare, and we don't do it in our vxlan equivalent, but I think we have seen cases where this might occur. 

We don't really want to block dataplane startup though, so we might need to start any retries in a new goroutine

Reply:
Hm... Good point. Should I add retrying to the existing vxlan function? I guess it wouldn't hurt, right?